ipyleaflet/leaflet.py

```diff
@@ -1172,8 +1172,6 @@ def add(self, layer):
 
         if isinstance(layer, dict):
             layer = basemap_to_tiles(layer)
-        elif hasattr(layer, 'as_leaflet_layer'):
```

Comment:
We should probably keep it here too though.

I wonder if we could implement `remove` and `substitute` too.

Reply:
Yes, let's do all that in follow-up PRs.